core/src/main/java/org/carbondata/core/carbon/datastore/block/Distributable.java

```diff
@@ -19,7 +19,7 @@
 /**
  * Abstract class which is maintains the locations of node.
  */
-public abstract class Distributable implements Comparable<Distributable> {
+public abstract class Distributable<T> implements Comparable<T> {
```

Comment:
having T extends Distributable will make better as we can limit the usage to the classes which are extending Distributable